docs/spec_draft.md

```diff
@@ -1586,6 +1587,48 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.select
+
+### Semantics
+
+Produces a `result` tensor where each element is selected from `on_true` or
+`on_false` tensor based on the value of the corresponding element of `pred`.
+More formally,
+`result[i0, ..., iR-1] = predicate[i0, ..., iR-1] ? on_true[i0, ..., iR-1] : on_false[i0, ..., iR-1]`,
+where `predicate[i0, ..., iR-1] = rank(pred) == 0 ? pred : pred[i0, ..., iR-1]`.
+
+### Inputs
+
+| Name       | Type                         |
+|------------|------------------------------|
+| `pred`     | tensor of type `i1`          |
+| `on_true`  | tensor of any supported type |
+| `on_false` | tensor of any supported type |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) Either `rank(pred)` $=$ `0` or `shape(pred)` $=$ `shape(on_true)`.
```

Comment:
I don't think that [shape_inference.cc](https://github.com/tensorflow/tensorflow/blob/38a677b555140410598603ff15abbe408b252e40/tensorflow/compiler/xla/service/shape_inference.cc#L3310-L3338) allows `pred` to have a different shape from `on_true`. Any specific reason to diverge from that?

Reply:
following up in the other thread.